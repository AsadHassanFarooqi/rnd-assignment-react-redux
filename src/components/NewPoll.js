import React, { useState } from 'react';
import { Col, Row } from 'react-bootstrap';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import Image from 'react-bootstrap/Image';
import { connect } from 'react-redux';
import { useNavigate, useRouteLoaderData } from 'react-router';
import { handleAddQuestion } from '../actions/questions';

const NewPoll = ({dispatch, authedUser}) => {
  const navigate = useNavigate();
    const [firstOption, setFirstOption] = useState("");
    const [secondOption, setSecondOption] = useState("");

    const handleFirstOptionChange = (e) => {
        const value = e.target.value;
        setFirstOption(value);
    };

    const handleSecondOptionChange = (e) => {
        const value = e.target.value;
        setSecondOption(value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        dispatch(handleAddQuestion(firstOption, secondOption));
        navigate("/home");
    };

  return (
    <div className="card">
      <Row>
        <Col xs="12" lg="12" className="d-flex align-items-center p-4">
          <Form className="w-100" onSubmit={handleSubmit}>
            <h2>Would you rather?</h2>
            <Form.Group className="mb-3">
              <Form.Label data-testid="firstOptionLabel">First option</Form.Label>
              <Form.Control type="text" placeholder="Enter first option" id="firstOption" onChange={handleFirstOptionChange} data-testid="firstOption" />
            </Form.Group>

            <Form.Group className="mb-3">
              <Form.Label data-testid="secondOptionLabel">Second option</Form.Label>
              <Form.Control type="text" placeholder="Enter second option" id="secondOption" onChange={handleSecondOptionChange} data-testid="secondOption" />
            </Form.Group>
            <Button variant="primary" type="submit" data-testid="submit-poll">
              Submit
            </Button>
          </Form>
        </Col>
      </Row>
    </div>
  )
}

const mapStateToProps = ({authedUser}) => ({
  authedUser
})

export default connect(mapStateToProps)(NewPoll);