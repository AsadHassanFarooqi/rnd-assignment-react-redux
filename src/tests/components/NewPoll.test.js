import React from "react";
import {Provider} from "react-redux";
import {fireEvent, render} from "@testing-library/react";
import {BrowserRouter} from "react-router-dom";

import { store } from "../../store";
import NewPoll from "../../components/NewPoll";


describe("NewPoll", () => {
  it("should render the component", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPoll/>
        </BrowserRouter>
      </Provider>
    );
    expect(view).toBeDefined();
    expect(view).toMatchSnapshot();
  });

  it.only("should display all elements", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPoll/>
        </BrowserRouter>
      </Provider>
    );

      const firstOptionLabelElement = view.getByTestId("firstOptionLabel");
      const firstOptionInputElement = view.getByTestId("firstOption");
      const secondOptionLabelElement = view.getByTestId("secondOptionLabel");
      const secondOptionInputElement = view.getByTestId("secondOption");
      const submitButtonElement = view.getByTestId("submit-poll");

      expect(firstOptionLabelElement.textContent).toBe("First option");
      expect(secondOptionLabelElement.textContent).toBe("Second option");
      expect(submitButtonElement.textContent).toBe("Submit");

      fireEvent.change(firstOptionInputElement, {target: {value: 'Texas'}});
      fireEvent.change(secondOptionInputElement, {target: {value: 'New Hampshire'}});
      expect(firstOptionInputElement.value).toBe("Texas");
      expect(secondOptionInputElement.value).toBe("New Hampshire");
  });
});
