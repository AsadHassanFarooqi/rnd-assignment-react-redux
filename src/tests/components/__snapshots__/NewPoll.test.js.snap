// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <div
          class="row"
        >
          <div
            class="d-flex align-items-center p-4 col-lg-12 col-12"
          >
            <form
              class="w-100"
            >
              <h2>
                Would you rather?
              </h2>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  data-testid="firstOptionLabel"
                >
                  First option
                </label>
                <input
                  class="form-control"
                  data-testid="firstOption"
                  id="firstOption"
                  placeholder="Enter first option"
                  type="text"
                />
              </div>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  data-testid="secondOptionLabel"
                >
                  Second option
                </label>
                <input
                  class="form-control"
                  data-testid="secondOption"
                  id="secondOption"
                  placeholder="Enter second option"
                  type="text"
                />
              </div>
              <button
                class="btn btn-primary"
                data-testid="submit-poll"
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <div
        class="row"
      >
        <div
          class="d-flex align-items-center p-4 col-lg-12 col-12"
        >
          <form
            class="w-100"
          >
            <h2>
              Would you rather?
            </h2>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                data-testid="firstOptionLabel"
              >
                First option
              </label>
              <input
                class="form-control"
                data-testid="firstOption"
                id="firstOption"
                placeholder="Enter first option"
                type="text"
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                data-testid="secondOptionLabel"
              >
                Second option
              </label>
              <input
                class="form-control"
                data-testid="secondOption"
                id="secondOption"
                placeholder="Enter second option"
                type="text"
              />
            </div>
            <button
              class="btn btn-primary"
              data-testid="submit-poll"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
